src/code/z_player_lib.c

```diff
@@ -1,153 +1,1021 @@
+}
+
+s32 func_80123F14(GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    return player->stateFlags1 & PLAYER_STATE1_800000;
+}
+
+s32 func_80123F2C(GlobalContext* globalCtx, s32 ammo) {
+    globalCtx->unk_1887C = ammo + 1;
+
+    return 1;
+}
+
+s32 Player_IsBurningStickInRange(GlobalContext* globalCtx, Vec3f* pos, f32 xzRange, f32 yRange) {
+    Player* this = GET_PLAYER(globalCtx);
+    Vec3f diff;
+    s32 pad;
+
+    if ((this->itemActionParam == PLAYER_AP_STICK) && (this->unk_B28 != 0)) {
+        Math_Vec3f_Diff(&this->meleeWeaponInfo[0].tip, pos, &diff);
+        return ((SQ(diff.x) + SQ(diff.z)) <= SQ(xzRange)) && (0.0f <= diff.y) && (diff.y <= yRange);
+    }
+
+    return false;
+}
+
+u8 Player_GetStrength(void) {
+    return D_801BFF98[(void)0, gSaveContext.playerForm];
+}
+
+u8 Player_GetMask(GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    return player->currentMask;
+}
+
+void Player_RemoveMask(GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    player->currentMask = PLAYER_MASK_NONE;
+}
+
+s32 Player_HasMirrorShieldEquipped(GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    return player->transformation == PLAYER_FORM_HUMAN && player->currentShield == PLAYER_SHIELD_MIRROR_SHIELD;
+}
+
+s32 Player_HasMirrorShieldSetToDraw(GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    return player->transformation == PLAYER_FORM_HUMAN && player->rightHandType == 8 &&
+           player->currentShield == PLAYER_SHIELD_MIRROR_SHIELD;
+}
+
+s32 Player_HoldsHookshot(Player* player) {
+    return player->itemActionParam == PLAYER_AP_HOOKSHOT;
+}
+
+s32 func_801240DC(Player* player) {
+    return Player_HoldsHookshot(player) && player->heldActor == NULL;
+}
+
+s32 func_80124110(Player* player, s32 actionParam) {
+    s32 temp_v0 = actionParam - PLAYER_AP_UNK_2;
+
+    if (((player->transformation != PLAYER_FORM_GORON) &&
+         ((actionParam - PLAYER_AP_UNK_2) > (PLAYER_AP_UNK_2 - PLAYER_AP_UNK_2))) &&
+        ((actionParam - PLAYER_AP_UNK_2) < (PLAYER_AP_SWORD_GREAT_FAIRY - PLAYER_AP_UNK_2))) {
+        return temp_v0;
+    }
+
+    return 0;
+}
+
+s32 func_80124148(Player* player) {
+    return func_80124110(player, player->itemActionParam);
+}
+
+s32 Player_ActionToMeleeWeapon(s32 actionParam) {
+    s32 sword = actionParam - PLAYER_AP_UNK_2;
+
+    if ((sword > (PLAYER_AP_UNK_2 - PLAYER_AP_UNK_2)) && (sword <= (PLAYER_AP_UNK_8 - PLAYER_AP_UNK_2))) {
+        return sword;
+    }
+    return 0;
+}
+
+s32 Player_GetMeleeWeaponHeld(Player* player) {
+    return Player_ActionToMeleeWeapon(player->itemActionParam);
+}
+
+s32 Player_HoldsTwoHandedWeapon(Player* player) {
+    // Relies on two-handed weapons to be contiguous
+    if ((player->itemActionParam >= PLAYER_AP_SWORD_GREAT_FAIRY) && (player->itemActionParam <= PLAYER_AP_STICK)) {
```

Comment:
This is the sort of thing that makes me want an actual macro to refer to this range. If you have to have this comment, it tells you the code is not clear. Much better to make the code clear in the first place.

Reply:
... and move the comment to the macro?
I don't think making a macro that would be used only in one place is worth enough. The function itself is already wrapping this dumb behavior in only one place

Reply:
Agreed with Anghelo. The behavior is already localized enough, the function is basically the macro in this case.